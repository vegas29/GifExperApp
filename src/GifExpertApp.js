    import React, { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

export const GifExpertApp = ({defaultCategories = []}) => {

    const [categories, setCategories] = useState(defaultCategories);

    return (
        <div className="container sm: mx-auto">
            <h1 className="text-white font-bold text-3xl md:text-4xl text-center border-b-2 pb-5">GifExpertApp</h1>
                <AddCategory
                    setCategories={setCategories}
                />

            {   
                
                categories.map(category => 
                    <GifGrid
                        key={category} 
                        category={category}
                    />
                )
            }
        </div>
    );
}
 
export default GifExpertApp;