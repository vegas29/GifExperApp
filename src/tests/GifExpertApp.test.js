/* eslint-disable testing-library/no-debugging-utils */
import { render, screen } from "@testing-library/react";
import GifExpertApp from "../GifExpertApp";

describe('Prueba sobre el componente <GifExpertApp />', () => {
    test('Deberia coincidir el snapshot', () => {
        const { container} = render( <GifExpertApp/> );
        expect( container ).toMatchSnapshot();

        screen.debug();
    });

    test('Debe contener el texto GifExpertApp', () => {
        render( <GifExpertApp/> );
        
        const h1 = screen.getByRole('heading', { level: 1 });
        expect( h1 ).toBeTruthy();
    });
});