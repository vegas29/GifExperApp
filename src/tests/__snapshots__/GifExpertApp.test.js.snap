// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba sobre el componente <GifExpertApp /> Deberia coincidir el snapshot 1`] = `
<div>
  <div
    class="container sm: mx-auto"
  >
    <h1
      class="text-white font-bold text-3xl md:text-4xl text-center border-b-2 pb-5"
    >
      GifExpertApp
    </h1>
    <form
      aria-label="form"
    >
      <input
        class="mt-5 rounded-xl bg-gray-100 focus:outline-none focus:border-2 focs:border-gray-200 py-2 px-5 text-xl"
        placeholder="Digite el nombre del gif a buscar"
        type="text"
        value=""
      />
    </form>
  </div>
</div>
`;
